raiden/network/proxies/token_network.py

```diff
@@ -176,7 +176,10 @@ def _new_channel_preconditions(
         if self.node_address == partner:
             raise SamePeerAddress('The other peer must not have the same address as the client.')
 
-        channel_exists = self.channel_exists_and_not_settled(
+        if not self.client.can_query_state_for_block(block_identifier):
```

Comment:
please, add a comment explaining why this is necessary.

Reply:
Actually this should never happen. Will raise an exception here.